src/depots/loader/DepotLoader.ts

```diff
@@ -0,0 +1,46 @@
+import { Depot, ProtonRequired } from 'src/depots/loader/Depot';
+import R2Error from 'src/model/errors/R2Error';
+import Game from 'src/model/game/Game';
+
+export default class DepotLoader {
+
+    public static DEPOT_DEFAULT_KEY = "default";
+
+    private static LOADED_DEPOTS: Map<string, Depot[]> = new Map();
+
+    private static load(): Map<string, Depot[]> {
+        const master = require('../master.json');
+        console.log("Master:", master);
+        const depotMap = new Map<string, Depot[]>();
+        for (let depotsKey in master.depots) {
+            // @ts-ignore
+            console.log("Depot key:", depotsKey);
+            depotMap.set(depotsKey, require(`../${master.depots[depotsKey]}`).depots as Depot[]);
+        }
+        this.LOADED_DEPOTS = depotMap;
+        return depotMap;
+    }
+
+    public static isProtonRequiredForDepot(game: Game, depotIdentifier: string): boolean {
+        if (!this.LOADED_DEPOTS.has(game.settingsIdentifier)) {
+            this.load();
+        }
+        const depots = this.LOADED_DEPOTS.get(game.settingsIdentifier);
+        if (depots === undefined) {
+            throw new R2Error(
+                `Unable to load depot for ${game.settingsIdentifier}`,
+                "This may be an issue with the manager. Report the issue in the appropriate discord server."
+            );
+        }
+        const foundDepotKey = Object.keys(depots).find(value => value === depotIdentifier);
```

Comment:
Couldn't this be simplified to just access the object via key and check if the result is undefined? Given that accessing a key that doesn't exist in javascript just returns `undefined` rather than failing.

I also think the `depots` variable should be type-hinted on line 28 already in a fashion that doesn't require casting later on, assuming it's possible.